tang.go

```diff
@@ -51,6 +54,50 @@ func (p TangPin) toConfig() (TangConfig, error) {
 	return c, nil
 }
 
+func roundup(size, blocksize int) int {
+	return size + blocksize - (size % blocksize)
+}
+
+// NIST SP 800-56 Concatenation Key Derivation Function (see section 5.8.1).
+func concatKDF(hash hash.Hash, z, s1 []byte, kdLen int) []byte {
```

Comment:
This could probably benefit from some unit tests?

Reply:
done